### DISCUSSION TITLE
Newbie here pls help me get started

### DISCUSSION THREAD

**Discussion by ivanvach** [tagyoureit/nodejs-poolController#650] [App and O/S Config Questions]
Followed the instructions strictly and installed 

Controller
REM
Web interface

Also installed PM2. 

PM2 shows all three running when I run

pm2 list

However I can't see anything at <IP address>:8080
Furthermore I can't make the API run at port 4200 getting errors
TypeError: Failed to fetch

Last but not least I see only port 22 open when I run a port scanner for the IP address. No other port shows open/available.

Is there any guide of how to troubleshoot and see what is wrong.

**Comment by Telemachus1977** [comment]
Run them individually to look for errors.

**Reply by ivanvach** [reply]
How do I do that? I’m copy/paste guy as far unix goes

**Reply by rstrouse** [reply]
Are you running an rdp or VNC session to connect to the pi headless or do you have a monitor connected?

Either way from a console window you can type `pm2 monit` to get a listing of the currently running processes as well as their console logs.

**Reply by ivanvach** [reply]
No monitor. I'm using putty ssh connection from my desktop.

**Reply by ivanvach** [reply]
Here is a screenshot after pm2 monit command

![Screenshot (512)](https://user-images.githubusercontent.com/43918172/207478997-3c676d0b-78e1-4fb6-a5f1-1f43160e7ee6.png)

**Reply by ivanvach** [reply]
I don't see any useful information to help troubleshooting. One suspicious thing is that all 3 processes use 0 CPU and 0 Memory.

**Reply by ivanvach** [reply]
Bottom right info window contains similar information for all 3 processes. One exception is that "watch and reload" is enabled for the first two and disabled for the last one.

**Reply by rstrouse** [reply]
Ok so the charset from putty is not ANSI and the terminal output from PM2 is.  You can get ANSI emulation from putty but I would suggest that you install xrdp and use remote desktop to connect to (I assume a Raspberry Pi).  It looks like you are running Windows is it Windows Home or Windows Pro?

To install xrdp on the pi run the following commands
`sudo apt update`
`sudo apt install xrdp`

This will install xrdp then you can simply use the Remote Desktop Connection in windows.

**Reply by ivanvach** [reply]
Done. I see now a desktop. What do I do now,

**Reply by rstrouse** [reply]
Open a console from there then run `pm2 monit`

**Reply by ivanvach** [reply]
Thanks will do. I have to run some errands now but will be back in 1 hour and stay as long as it takes to get this resolved

**Reply by rstrouse** [reply]
I am guessing you missed something in the install.

Here is a quick start guide that includes the stuff that should be installed for a Pi.
https://github.com/tagyoureit/nodejs-poolController/wiki/Quick-Start-Guide

**Comment by ivanvach** [comment]
Well pm2 monit produced prettier screen on the desktop, however the information is same. All 3 processes are running with 0 Mem and 0 CPU.
Did review the quick start and that is exactly what I did. In addition to that I went to config.json to enable REM as it was disabled by default.
I also installed pm2 that is working obviously.

**Reply by ivanvach** [reply]
![Screenshot (514)](https://user-images.githubusercontent.com/43918172/207509501-552afd97-a57d-4971-a528-b01a066faca7.png)

**Reply by Telemachus1977** [reply]
start by going into each directory individually (start with REM since it generates a web interface on its own) and stype npm start.  

Watch what happens.

**Reply by ivanvach** [reply]
All 3  do work correctly when started individually.
Looks like the problem is pm2 .
Any ideas or advise how to troubleshoot and fix it?

**Reply by Telemachus1977** [reply]
Which Pi are you using?  With a zero w, it crashed when trying to compile all of them at the same time.

**Reply by Telemachus1977** [reply]
try installing SCREEN.  run one, then put the terminal in the background and run another. Then put that in the background and run the third.

You can also setup PM2 to run without compiling.

**Reply by ivanvach** [reply]
Yes, I managed to run them simultaneously as you suggest.

**Reply by Telemachus1977** [reply]
and its all working fine?

**Reply by ivanvach** [reply]
Yes all working except for PM2.
If I have them all working and reboot the Pi than noting works.

**Comment by ivanvach** [comment]
Furthermore when running 'pm2 logs' I get the following error repeating over and over again:

Error caught while calling pidusage
PM2      | TypeError: One of the pids provided is invalid
PM2      |     at get (/home/ivan/.nvm/versions/node/v12.18.3/lib/node_modules/pm2/node_modules/pidusage/lib/stats.js:78:23)
PM2      |     at pidusage (/home/ivan/.nvm/versions/node/v12.18.3/lib/node_modules/pm2/node_modules/pidusage/index.js:30:5)
PM2      |     at Object.getMonitorData (/home/ivan/.nvm/versions/node/v12.18.3/lib/node_modules/pm2/lib/God/ActionMethods.js:60:5)
PM2      |     at tasks (/home/ivan/.nvm/versions/node/v12.18.3/lib/node_modules/pm2/lib/Worker.js:150:9)
PM2      |     at Domain.<anonymous> (/home/ivan/.nvm/versions/node/v12.18.3/lib/node_modules/pm2/lib/Worker.js:193:7)
PM2      |     at Domain.run (domain.js:371:14)
PM2      |     at Timeout.wrappedTasks [as _onTimeout] (/home/ivan/.nvm/versions/node/v12.18.3/lib/node_modules/pm2/lib/Worker.js:192:7)
PM2      |     at listOnTimeout (internal/timers.js:549:17)
PM2      |     at processTimers (internal/timers.js:492:7)

**Reply by Telemachus1977** [reply]
You might have to poke around on the pm2 forums/reddit subchannels

**Reply by tagyoureit** [reply]
Did you by chance change the node version (through NVM) after installed PM2?  You should run the commands on this page to make sure PM2 is updated and is accounting for the right node version:  https://pm2.keymetrics.io/docs/usage/update-pm2/

**Comment by ivanvach** [comment]
This PM2 thing gets out of control. More problems emerge as I try to follow the upgrade instructions. I'm not a quitter but I don't see much reward pursuing this since this is for power outages only and my power is relatively steady.

**Reply by Telemachus1977** [reply]
There are certainly other ways to autostart the system on startup.
